### PR TITLE
fix(manual_find_map and manual_filter_map): check clone method

### LINKED ISSUE #8920: Lint request: filter(is_some).map(unwrap)
### What it does

Lint about usage of `.filter(/* ... */).map(/* ... */)` calls that could be simplified to `.filter_map(/* ... */)`.

### Lint Name

filter_then_map

### Category

_No response_

### Advantage

- Removes an unnecessary `.unwrap()`
  - … or `.expect()`
  - … or `.unwrap_or*` with a never-used default value / expression
- Makes the code easier to read and shorter

### Drawbacks

_No response_

### Example

```rust
struct Foo {
    field: Option<String>,
}

fn main() {
    let vec = vec![Foo {
        field: Some("value".to_owned()),
    }];
    let iter = vec
        .iter()
        .filter(|f| f.field.is_some())
        .map(|f| f.field.clone().unwrap());

    for field in iter {
        dbg!(field);
    }
}
```

Could be written as:

```rust
struct Foo {
    field: Option<String>,
}

fn main() {
    let vec = vec![Foo {
        field: Some("value".to_owned()),
    }];
    let iter = vec.iter().filter_map(|f| f.field.clone());

    for field in iter {
        dbg!(field);
    }
}
```

### PR DESCRIPTION
**Pull request by kyoto7250**
close #8920 

Added conditional branching when the clone method is used.

Thank you in advance.

---

changelog: check `clone()` and other variant preserving methods in [`manual_find_map`] and [`manual_filter_map`]

### CODE REVIEW

**Comment by kyoto7250** [comment]
@Alexendoo 

I commited your suggestion.
Could you review again?

File: clippy_lints/src/methods/filter_map.rs [hunk 1]
```diff
@@ -171,3 +179,16 @@ pub(super) fn check<'tcx>(
...
 }
+
+fn acceptable_methods(method: &PathSegment<'_>) -> bool {
+    let methods: [Symbol; 6] = [
+        sym::clone,
+        sym::as_ref,
+        sym!(as_deref),
+        sym!(as_mut),
+        sym!(as_deref_mut),
+        sym!(to_owned),
```
Comment (Alexendoo) [inline clippy_lints/src/methods/filter_map.rs:190]: `cloned` and `copied` should also be applicable

Comment (kyoto7250) [reply clippy_lints/src/methods/filter_map.rs:192]: Sorry, I missed.

I fixed by [this commit](https://github.com/rust-lang/rust-clippy/pull/8930/commits/42cf98553a72f4b6cdac2c4c6378121d30f3bf5f), thank you!

**Review by Alexendoo — APPROVED** [review]
nice, thanks!

**Pull request merged**